[Uid] Default to UuidV7 when using UuidFactory (#21408)
symfony/symfony-docs: components/uid.rst

    $uuid = Uuid::fromBase58('TuetYWNHhmuSQ3xPoVLv9M');
    $uuid = Uuid::fromRfc4122('d9e7a184-5d5b-11ea-a62a-3499710062d0');

You can also use the ``UuidFactory`` to generate UUIDs. Inject the factory in
your services and use it as follows:

    namespace App\Service;

    use Symfony\Component\Uid\Factory\UuidFactory;

    class FooService
    {
        public function __construct(
            private UuidFactory $uuidFactory,
        ) {
        }

        public function generate(): void
        {
            $uuid = $this->uuidFactory->create();

>             $nameBasedUuid = $this->uuidFactory->nameBased(/** ... */);

Reviewer: The `nameBased()`, `timeBased()` and `randomBased()` methods don't return a UUID directly, but return a factory instead on which we then have to call `create()`. So this example would have to look like this:

```suggestion
            $nameBasedUuid = $this->uuidFactory->nameBased(/** ... */)->create();
```

Reviewer 2: `->create($name)` actually (the argument of nameBased is optional if the factory is configured with a default namespace)

>             $randomBasedUuid = $this->uuidFactory->randomBased();

Reviewer: ```suggestion
            $randomBasedUuid = $this->uuidFactory->randomBased()->create();
```

>             $timestampBased = $this->uuidFactory->timeBased();

Reviewer: ```suggestion
            $timestampBased = $this->uuidFactory->timeBased()->create();
```

    $uuid = Uuid::fromBase58('TuetYWNHhmuSQ3xPoVLv9M');
    $uuid = Uuid::fromRfc4122('d9e7a184-5d5b-11ea-a62a-3499710062d0');

You can also use the ``UuidFactory`` to generate UUIDs. Inject the factory in
your services and use it as follows:

    namespace App\Service;

    use Symfony\Component\Uid\Factory\UuidFactory;

    class FooService
    {
        public function __construct(
            private UuidFactory $uuidFactory,
        ) {
        }

        public function generate(): void
        {
            $uuid = $this->uuidFactory->create();

            $randomBasedUuid = $this->uuidFactory->randomBased()->create();
            // $namespace can be omitted if a default is configured in the factory (see below)

>             $nameBasedUuid = $this->uuidFactory->nameBased($namespace)->create();

Reviewer: ```suggestion
            $nameBasedUuid = $this->uuidFactory->nameBased($namespace)->create($name);
```

Author: Sorry I missed this. Added. Thanks!

    $uuid = Uuid::fromBase58('TuetYWNHhmuSQ3xPoVLv9M');
    $uuid = Uuid::fromRfc4122('d9e7a184-5d5b-11ea-a62a-3499710062d0');

You can also use the ``UuidFactory`` to generate UUIDs. By default, this factory
uses UUIDv7 for default and time-based UUIDs, UUIDv5 for name-based UUIDS, and

> UUIDv4 for random-based UUIDs, but you can configure this behavior::

Reviewer: We should probably update the config example to use other versions than 7 for the default and time based UUID version.

Author: Good idea! I submitted some changes. Thanks.

            $container->extension('framework', [
                'uid' => [
                    'default_uuid_version' => 6,
                    'name_based_uuid_version' => 3,
                    'name_based_uuid_namespace' => '6ba7b810-9dad-11d1-80b4-00c04fd430c8',
                    'time_based_uuid_version' => 6,
                    'time_based_uuid_node' => 121212121212,
                ],
            ]);
        };

.. versionadded:: 7.4

    Starting from Symfony 7.4, the default version for both UUIDs and time-based

>     UUIDs is UUIDv7. In previous versions, the default was UUIDv6.

Reviewer 3: ```suggestion
    UUIDs is v7. In previous versions, the default was v6.
```
